.github/workflows/ci.yml

```diff
@@ -2,374 +2,208 @@ name: Keycloak CI
-          exit $TEST_RESULT
-
-      - name: Analyze Test and/or Coverage Results
-        uses: runforesight/foresight-test-kit-action@v1.3.0
-        if: always() && github.repository == 'keycloak/keycloak'
-        with:
-          api_key: ${{ secrets.FORESIGHT_API_KEY }}
-          test_format: JUNIT
-          test_framework: JUNIT
-          test_path: 'testsuite/integration-arquillian/tests/base/target/surefire-reports/*.xml'
-
-      - name: Base test reports
-        uses: actions/upload-artifact@v3
-        if: failure()
-        with:
-          name: reports-${{ matrix.server }}-base-tests-${{ matrix.tests }}
-          retention-days: 14
-          path: reports-${{ matrix.server }}-base-tests-${{ matrix.tests }}.zip
-          if-no-files-found: ignore
+          TESTS=`testsuite/integration-arquillian/tests/base/testsuites/suite.sh jdk`
+          echo "Tests: $TESTS"
+          ./mvnw install -nsu -B -Pauth-server-quarkus -Pdb-${{ matrix.db }} -Dtest=$TESTS -pl testsuite/integration-arquillian/tests/base | misc/log/trimmer.sh
 
-  test-fips:
-    name: Base testsuite (fips)
+  new-store-integration-tests:
+    name: New Store IT
     needs: build
     runs-on: ubuntu-latest
-    timeout-minutes: 45
+    timeout-minutes: 100
     strategy:
       matrix:
-        server: ['bcfips-nonapproved-pkcs12']
-        tests: ['group1', 'group2']
+        db: ['chm', 'hot-rod', 'jpa']
       fail-fast: false
     steps:
       - uses: actions/checkout@v3
-        with:
-          fetch-depth: 2
 
-      - name: Cache Maven packages
-        uses: actions/cache@v3
-        with:
-          path: ~/.m2/repository
-          key: cache-2-${{ runner.os }}-m2-${{ hashFiles('**/pom.xml') }}
-          restore-keys: cache-1-${{ runner.os }}-m2
+      - id: integration-test-setup
+        name: Integration test setup
+        uses: ./.github/actions/integration-test-setup
 
-      - name: Download built keycloak
-        id: download-keycloak
-        uses: actions/download-artifact@v3
-        with:
-          path: ~/.m2/repository/org/keycloak/
-          name: keycloak-artifacts.zip
+      - name: Run base tests
+        run: |
+          declare -A PARAMS
+          PARAMS["chm"]="-Pmap-storage -Dpageload.timeout=90000"
+          PARAMS["hot-rod"]="-Pmap-storage,map-storage-hot-rod -Dpageload.timeout=90000"
+          PARAMS["jpa"]="-Pmap-storage,map-storage-jpa -Dpageload.timeout=90000"
 
-      # - name: List M2 repo
-      #   run: |
-      #     find ~ -name *dist*.zip
-      #     ls -lR ~/.m2/repository
+          TESTS=`testsuite/integration-arquillian/tests/base/testsuites/suite.sh database`
+          echo "Tests: $TESTS"
+          ./mvnw install -nsu -B -Pauth-server-quarkus ${PARAMS["${{ matrix.db }}"]} -Dtest=$TESTS -pl testsuite/integration-arquillian/tests/base | misc/log/trimmer.sh
+
+  legacy-store-integration-tests:
+    name: Legacy Store IT
+    needs: build
+    runs-on: ubuntu-latest
+    timeout-minutes: 100
```

Comment:
Would the job timeouts be tuned in the final phase of this PR? This seems to be a _very_ generous timeout for a 10-minute job

Reply:
Changed to 30 min

Reply:
Thank you, resolving this